File: src/main/java/dataaccess/mongodb/databaseConnection.js
const {MongoClient} = require('mongodb');

async function main() {

    const uri = "mongodb+srv://tasnimreza:<dbtemppassword>@atlascluster.tijckuy.mongodb.net/?retry" +
        "Writes=true&w=majority&appName=AtlasCluster"


    const client = new MongoClient(uri);

        try {
            await client.connect();
            await listDatabases(client);
        } catch (e) {
            console.error(e);
        } finally {
            await client.close();
        }
}

main().catch(console.error);

async function listDatabases(client){
    let databasesList = await client.db().admin().listDatabases();

    console.log("Databases:");
    databasesList.databases.forEach(db => console.log(` - ${db.name}`));
}

File: src/main/java/dataaccess/mongodb/DatabaseConnection.java
package dataaccess.mongodb;

import com.mongodb.ConnectionString;
import com.mongodb.MongoClientSettings;
import com.mongodb.MongoException;
import com.mongodb.ServerApi;
import com.mongodb.ServerApiVersion;
import com.mongodb.client.MongoClient;
import com.mongodb.client.MongoClients;
import com.mongodb.client.MongoCollection;
import com.mongodb.client.MongoDatabase;
import org.bson.Document;

import java.util.ArrayList;
import java.util.List;

public class DatabaseConnection {
    public static void main(String[] args) {
        String connectionString = "mongodb+srv://tasnimreza:dbtestpass@cluster0.vlnfmzu.mongodb.net/?appName=Cluster0";

        ServerApi serverApi = ServerApi.builder()
                .version(ServerApiVersion.V1)
                .build();

        MongoClientSettings settings = MongoClientSettings.builder()
                .applyConnectionString(new ConnectionString(connectionString))
                .serverApi(serverApi)
                .build();

        // Create a new client and connect to the server
        try (MongoClient mongoClient = MongoClients.create("mongodb+srv://tasnimreza:dbtestpass@cluster0.vlnfmzu.mongodb.net/?retryWrites=true&w=majority&appName=Cluster0")) {
            try {
                // Send a ping to confirm a successful connection
                MongoDatabase database = mongoClient.getDatabase("sample_mflix");
                MongoCollection<Document> collection = database.getCollection("comments");
                database.runCommand(new Document("ping", 1));
                System.out.println("Pinged your deployment. You successfully connected to MongoDB!");
                List<Document> databases = mongoClient.listDatabases().into(new ArrayList<>());
                databases.forEach(db -> System.out.println(db.toJson()));

            } catch (MongoException e) {
                e.printStackTrace();
            }
            finally {
                System.out.println("meepmeep");
            }
        }
    }
}
